astropy/visualization/wcsaxes/tests/test_misc.py

```diff
@@ -507,17 +509,25 @@ def test_set_labels_with_coords(ignore_matplotlibrc, frame_class):
         assert ax.coords[i].get_axislabel() == labels[i]
 
 
-# The bounding box calculation is very dependent on Matplotlib versions.
-@pytest.mark.skipif('not MATPLOTLIB_EQ_33')
-def test_bbox_size():
-    # Test for the size of a WCSAxes bbox
+@pytest.mark.parametrize('atol', [0.2, 1.0e-8])
+def test_bbox_size(atol):
+    """Test for the size of a WCSAxes bbox with 'tight' layout.
+    """
+    extents = [11.38888888888889, 3.5]
+    # Upper bounding box limits changed between Matplotlib major versions.
+    if MATPLOTLIB_LT_30:
+        extents += [579.5, 435.5]
+    else:
+        extents += [576.0, 432.0]
+
     fig = plt.figure()
     ax = WCSAxes(fig, [0.1, 0.1, 0.8, 0.8])
     fig.add_axes(ax)
     fig.canvas.draw()
     renderer = fig.canvas.renderer
     ax_bbox = ax.get_tightbbox(renderer)
-    assert np.allclose(ax_bbox.x0, 11.38888888888889)
-    assert np.allclose(ax_bbox.x1, 576)
-    assert np.allclose(ax_bbox.y0, 3.5)
-    assert np.allclose(ax_bbox.y1, 432)
+
+    # Enforce strict test only with reference Freetype version
+    if atol < 0.1 and not FREETYPE_261:
```

Comment:
Instead of running it twice with different `atol`, why not run it once but set the `atol` according to `FREETYPE_261`?

Reply:
Yes, as noted in https://github.com/astropy/astropy/pull/11022#issue-518786940 there are multiple redundant testing levels now that may not all be needed. Same perhaps for testing the other case as `xfail` rather than skipping it – I think yet other Freetype versions _might_ fail on the stricter tolerance, but don't necessarily need to.
From a practical pov ~0.2 pt accuracy would probably be plenty for all purposes, but otoh for known mpl+ft versions we obviously can expect more precise results.
For consistency with #11022/v4.3 I'd probably leave it as is, but in the interest of keeping testing load low this can well do with a single run.